src/postprocessors/RankTwoTensorInvariantPostprocessor.C

```diff
@@ -0,0 +1,65 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "RankTwoTensorInvariantPostprocessor.h"
+
+registerMooseObject("DeerApp", RankTwoTensorInvariantPostprocessor);
+
+InputParameters RankTwoTensorInvariantPostprocessor::validParams() {
+  InputParameters params = GeneralPostprocessor::validParams();
+  params.addClassDescription(
+      "Computes an invaraint of a symmetric rank two "
+      "tensor given a set of postprocessors"
+      " representing tensorial components. This class "
+      "leverages MOOSE RankTwoScalarTools::invariantOptions()");
+  params.addRequiredParam<PostprocessorName>(
+      "rank_two_tensor_base_name",
+      "The base name of the rank_two_tensor postprocessors from which the "
+      "invariant shall to be computed.");
+  params.addParam<MooseEnum>(
+      "invariant", RankTwoScalarTools::invariantOptions(),
+      "Type of inveriant (see RankTwoScalarTools::invariantOptions for "
```

Comment:
Ehh, why don't you copy over some of the common options into the docstring.  At least the ones that I'll use and forget continually.